scripts/pmbuild.py

```diff
@@ -753,6 +753,43 @@ def replace_user_vars(arg, config):
     return arg
 
 
+def fill_user_vars(raw_config, config):
+    ignored_vars = [
+        "input_file",
+        "output_file",
+        "target_path",
+        "target_name",
+        "export_args",
+        "vs_latest",
+        "target_name",
+        "container_file",
+        "windows_sdk_version",
+        "teamid",
+        "cwd"
+    ]
+
+    idx_end = 0
+    idx_start = 0
+
+    while idx_end != -1 and idx_start != -1:
+        idx_start = raw_config.find("%{",idx_end)
+        if idx_start != -1:
+            idx_end = raw_config.find("}",idx_start)
+            if idx_end != -1:
+                var_name = raw_config[idx_start+2:idx_end]
+                if var_name in ignored_vars:
+                    continue
+
+                value = ""
+                if "user_vars" not in config.keys() or var_name not in config["user_vars"]:
+                    print( "User var '"+var_name+"' not defined")
+                if "user_vars" in config.keys() and var_name in config["user_vars"]:
+                    value = config["user_vars"][var_name]
+                raw_config = raw_config[:idx_start] + value + raw_config[idx_end+1:]
+                idx_end - 3 # remove offset of 3 characters %{}
```

Comment:
I made some mistake here, need to calculate it properly and assign the value correctly